saleor/warehouse/management.py

```diff
@@ -307,14 +343,17 @@ def decrease_stock(order_lines_info: Iterable["OrderLineData"], update_stocks=Tr
         quantity_allocation_for_stocks[allocation["stock"]] += allocation[
             "quantity_allocated__sum"
         ]
-
     if update_stocks:
         _decrease_stocks_quantity(
             order_lines_info,
             variant_and_warehouse_to_stock,
             quantity_allocation_for_stocks,
         )
 
+        for stock in stocks:
+            if stock.available_quantity() <= 0:
```

Comment:
In your soultion we execute databse query for each line separately. 
```suggestion
        for stock in stocks.annotate_available_quantity():
            if stock.available_quantity<= 0:
```

Reply:
I can't just conduct annotations because `available_quantity` is a method of the `Stock` object.